packages/runtime/src/decorators/middlewares.ts

```diff
@@ -0,0 +1,125 @@
+import 'reflect-metadata';
+export type ExpressMiddleware = (req: any, res: any, next: any) => Promise<any>;
+export type KoaMiddleware = (ctx: any, next: any) => Promise<any>;
+export type HapiMiddlewareBase = (request: any, h: any) => Promise<any>;
+export type HapiMiddlewareSimple = HapiMiddlewareBase | { method: HapiMiddlewareBase; assign?: string; failAction?: HapiMiddlewareBase | string };
```

Comment:
Would prefer if we could try to type this properly as they are visible

Reply:
I'll try to do it, though that means I'll have to add other dependencies (the typing of `express`, `koa` and `hapi`) to the project.
While this means we'll need to keep up-to-date with those dependencies too, at least we'll know when/if they introduce non-retro compatible changes in their middleware definitions (not that I expect those to happen, but in theory they could do something very asinine like, e.g., putting `next` as 1st parameter instead of the `3rd` for the `express` middleware)

Reply:
That's probably worth the
a) DX
b) as you mentioned, the chance to detect potential issues earlier

Reply:
TBH, I wouldn't mind to have middlewares not typed (using `any`). I think it's better solution than having dependencies on express, koa and hapi types and maintaining those dependencies...